app/services/user_silencer.rb

```diff
@@ -23,7 +23,9 @@ def self.was_silenced_for?(post)
 
   def silence
     hide_posts unless @opts[:keep_posts]
-    unless @user.silenced_till.present?
+    if @user.silenced_till.present?
```

Comment:
Could also easily be an early return :) `return false if @user.silenced_till.present?`